libs/beacon/src/beacon_manager.cpp

```diff
@@ -27,31 +27,21 @@
 namespace fetch {
 namespace dkg {
 
-bn::G2 BeaconManager::zeroG2_;
-bn::Fr BeaconManager::zeroFr_;
-bn::G2 BeaconManager::group_g_;
-bn::G2 BeaconManager::group_h_;
-
 constexpr char const *LOGGING_NAME = "BeaconManager";
+// To set elliptic curve generator in static verify function
+constexpr char const *generator_string = "Fetch.ai Elliptic Curve Generator G";
 
 BeaconManager::BeaconManager(CertificatePtr certificate)
-  : certificate_{std::move(certificate)}
+  : group_g_{generator_string}
+  , group_h_{"Fetch.ai Elliptic Curve Generator H"}
```

Comment:
It's a bit odd here that generator_string is a constexpr for g, but h pastes it in directly. Also, in my mind the beacon manager is really just trying to hold keys, having the generator string here and in a few other places makes it unclear what is responsible for setting these.

I think in mcl_dkg it could be something like

```
void SetGenerator(Generator &generator_g, std::string string_to_hash = "Fetch.ai Elliptic Curve Generator G")
{
  assert(!string_to_hash.empty());
  bn::hashAndMapToG2(generator_g, string_to_hash);
}
```

And here it would be cleaner if it were

```
group_g_{mcl::DefaultGeneratorG()}
```